src/v3/src/util/theme.ts

```diff
@@ -10,116 +10,429 @@
+    set(theme, 'typography.h1.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h1.fontSize', mergedTokens.TypographySizeHeading1);
+    set(theme, 'typography.h1.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h1.lineHeight', mergedTokens.TypographyLineHeightHeading1);
+    set(theme, 'typography.h2.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h2.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h2.fontSize', mergedTokens.TypographySizeHeading2);
+    set(theme, 'typography.h2.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h2.lineHeight', mergedTokens.TypographyLineHeightHeading2);
+    set(theme, 'typography.h3.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h3.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h3.fontSize', mergedTokens.TypographySizeHeading3);
+    set(theme, 'typography.h3.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h3.lineHeight', mergedTokens.TypographyLineHeightHeading3);
+    set(theme, 'typography.h4.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h4.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h4.fontSize', mergedTokens.TypographySizeHeading4);
+    set(theme, 'typography.h4.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h4.lineHeight', mergedTokens.TypographyLineHeightHeading4);
+    set(theme, 'typography.h5.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h5.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h5.fontSize', mergedTokens.TypographySizeHeading5);
+    set(theme, 'typography.h5.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h5.lineHeight', mergedTokens.TypographyLineHeightHeading5);
+    set(theme, 'typography.h6.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h6.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h6.fontSize', mergedTokens.TypographySizeHeading6);
+    set(theme, 'typography.h6.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h6.lineHeight', mergedTokens.TypographyLineHeightHeading6);
+    set(theme, 'typography.kbd.fontFamily', mergedTokens.TypographyFamilyMono);
+    set(theme, 'typography.overline.lineHeight', mergedTokens.TypographyLineHeightOverline);
+    set(theme, 'typography.subtitle1.fontFamily', mergedTokens.TypographyFamilyBody);
+    set(theme, 'typography.subtitle2.fontFamily', mergedTokens.TypographyFamilyBody);
+  }
+  return mergeThemes(theme, {
+    components: {
+      MuiInputLabel: {
+        styleOverrides: {
+          root: {
+            wordBreak: 'break-word',
+            whiteSpace: 'normal',
+          },
+        },
+      },
+      MuiButton: {
```

Comment:
Had a tough time untangling design tokens from component overrides. `MuiButton` is the most visible issue but needs testing to find other occurrences.

Reply:
I'll check on SauceLabs, but CSS contains nothing unusual. It should be applied by the conditional `ownerState.variant` spread styles...

```ts
theme.components?.MuiButton?.styleOverrides?.root = ({ownerState, theme: t}) => ({
  ...(ownerState.variant === 'primary' && {
    '&:hover': {
      backgroundColor: t.palette.primary.dark,
    },
  }),
})
```